src/ray/ray_impl.rs

```diff
@@ -106,6 +107,40 @@ impl<T: Scalar + Copy, const D: usize> Ray<T, D> {
         self.ray_intersects_aabb(aabb)
     }
 
+    /// Intersect [`Aabb`] by [`Ray`]
+    /// Returns slice of intersections, two numbers `T`
+    /// where the first number is the distance from [`Ray`] to the nearest intersection point
+    /// and the second number is the distance from [`Ray`] to the farthest intersection point
+    /// 
+    /// If there are no intersections, it returns infinity for both distances
+    pub fn intersection_slice_for_aabb(&self, aabb: &Aabb<T, D>) -> (T, T)
+    where
+        T: ClosedSub + ClosedMul + ClosedAdd + Signed + Zero + PartialOrd + SimdPartialOrd + 
+        FromPrimitive + Copy + Scalar + Float,
+    {
+        // https://iquilezles.org/articles/intersectors/
+        let mut n = self.origin.coords - aabb.center().coords;
+        n.component_mul_assign(&self.inv_direction);
+
+        let k = self.inv_direction.abs().component_mul(&aabb.half_size());
+        let t1 = -n - k;
+        let t2 = -n + k;
+        
+        let entry_distance = t1.iter().skip(1).fold(
+            t1[0], |acc, x| -> T { Float::max(*x, acc) }
```

Comment:
```suggestion
            t1[0], |acc, x| -> T { fast_max(*x, acc) }
```

RealField also has a min and max, we also have a fast_min and fast_max in `utils` for all scalars

Reply:
why "fast" max? Did you run benchmarks to approve this?